src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -714,6 +720,107 @@ private function getMergeFormData(int $orderId, int $shipmentId): array
         ];
     }
 
+    #[AdminSecurity("is_granted('update', 'AdminOrders')", redirectRoute: 'admin_orders_view', redirectQueryParamsToKeep: ['orderId'], message: 'You do not have permission to edit this.')]
+    public function splitShipmentAction(int $orderId, int $shipmentId, Request $request): RedirectResponse
+    {
+        $data = $request->get('split_shipment', []);
+        $carrier = isset($data['carrier']) ? (int) $data['carrier'] : null;
+
+        $productsRaw = $data['products'] ?? [];
```

Comment:
Did you have the chance to try this syntax?

```suggestion
        $carrier = $request->getInt('split_shipment.carrier', null);
        $productsRaw = $request->get('split_shipment.products', []);
```

Reply:
not possible sorry